tests/test_merge.py

```diff
@@ -145,3 +149,89 @@ def test_get_average_frame_count(samples: str, count: float) -> None:
+                perf_text="python-123/123;_PyObject_Call_Prepend;_PyObject_FastCallDict;_PyFunction_FastCallDict 2\n"
+                "python-123/123;_start;__libc_start_main;main;Py_Main;PyRun_SimpleFileExFlags 3\n"
+                "python-123/123;entry_SYSCALL_64_[k] 1",
+                process_text="python-123/123;[Profiling error: exception CalledProcessError] 1",
+            ),
+            ";python;[Profiling error: exception CalledProcessError];_PyObject_Call_Prepend;_PyObject_FastCallDict;"
+            "_PyFunction_FastCallDict 2\n"
+            ";python;[Profiling error: exception CalledProcessError];_start;__libc_start_main;main;Py_Main;"
+            "PyRun_SimpleFileExFlags 3\n"
+            ";python;[Profiling error: exception CalledProcessError];entry_SYSCALL_64_[k] 1",
+            id="3perf_1pyspy-error",
+        ),
+        pytest.param(
+            dict(
+                perf_text="java-123/123;(/opt/java/lib/libjvm.so);__vdso_gettimeofday;(vdso);apic_timer_interrupt_[k];"
+                "smp_apic_timer_interrupt_[k] 4",
+                process_text="java-123/123;[Profiling skipped: async-profiler is already loaded] 1",
+            ),
+            ";java;[Profiling skipped: async-profiler is already loaded];(/opt/java/lib/libjvm.so);__vdso_gettimeofday;"
+            "(vdso);apic_timer_interrupt_[k];smp_apic_timer_interrupt_[k] 4",
+            id="1perf_1java-skipped",
+        ),
+        pytest.param(
+            dict(
+                perf_text="python-123/123;EMPTY 0",
+                process_text="python-123/123;[Profiling error: exception CalledProcessError] 1",
+            ),
+            "",
+            id="empty-perf_1pyspy-error",
+        ),
+        pytest.param(
+            dict(
+                perf_text="python-123/123;Py_Main;PyRun_SimpleFileExFlags 5\n"
+                "java-456/456;[unknown];pthread_getname_np;do_syscall_64_[k] 3",
+                process_text="python-123/123;[Profiling error: exception CalledProcessError] 1\n"
+                "java-456/456;[Profiling skipped: profiled-oom] 1",
+            ),
+            ";java;[Profiling skipped: profiled-oom];[unknown];pthread_getname_np;do_syscall_64_[k] 3\n"
+            ";python;[Profiling error: exception CalledProcessError];Py_Main;PyRun_SimpleFileExFlags 5",
+            id="2pids_1perf-each_1java_1pyspy",
+        ),
+    ],
+)
+def test_merge_profiles_onto_errors(input_dict: Dict[str, str], expected: str) -> None:
+    enrichment_options = EnrichmentOptions(
+        profile_api_version=None,
+        container_names=False,
+        application_identifiers=False,
+        application_identifier_args_filters=[],
+        application_metadata=False,
+    )
+    metadata: Metadata = dict(profiling_mode="cpu")
+    metrics = Metrics(cpu_avg=1.0, mem_avg=1.0)
+    perf_pid_to_profiles = parse_profiles_text(input_dict["perf_text"])
+    process_profiles = parse_profiles_text(input_dict["process_text"])
+    outcome_split = sorted(
```

Comment:
Why `sorted`? You rely on `#` being first or I didn't understand what's going on here :/

I think it's best to do it explicit, maybe as I suggested in the prev comment?

Reply:
I wanted to ensure predictable order of the lines in the merged trace to be able to compare outlines.
I agree it was overkill to sort all the lines - the lines of trace are processed in order.
What was bugging me was iteration order of pids in `ProcessToProfileData` dictionaries.  

Now I know it's not a problem since Python 3.7.
I have missed the change: dictionaries became insertion-ordered by standard in 3.7.
Will replace with a simpler solution.

Reply:
Ahhh I now understand what you tried to resolve :) If it was commented I'd suggest we just keep it haha, but yeah since we run Python 3.7+ nowadays (and I'm going to remove the 3.6 testing) and the solution now is okay as well.